providers/google.go

```diff
@@ -294,3 +357,25 @@ func FindAvaliablePort(redirectURIs []string) (*url.URL, net.Listener, error) {
 	}
 	return nil, nil, fmt.Errorf("failed to start a listener for the callback from the OP, got %w", lnErr)
 }
+
+func configCookieHandler() (*httphelper.CookieHandler, error) {
+	// I've been unable to determine a scenario in which setting a hashKey and blockKey
+	// on the cookie provide protection in the localhost redirect URI case. However I
+	// see no harm in setting it.
+	hashKey := make([]byte, 64)
+	if _, err := io.ReadFull(rand.Reader, hashKey); err != nil {
+		return nil, fmt.Errorf("failed to generate random keys for cookie storage")
+	}
+	blockKey := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, blockKey); err != nil {
+		return nil, fmt.Errorf("failed to generate random keys for cookie storage")
+	}
+
+	// OpenPubkey uses a localhost redirect URI to receive the authcode
+	// from the OP. Localhost redirects use http not https. Thus, we should
+	// not set these cookies as secure-only. This should be changed if
+	// OpenPubkey added support for non-localhost redirect URIs.
+	// WithUnsecure() is equivalent to not setting the 'secure' attribute
+	// flag in an HTTP Set-Cookie header (see https://http.dev/set-cookie#secure)
```

Comment:
I agree, this is on my mental list of things to do. I just don't have an immediate use for it for it so I haven't done it. If we got OpenPubkey running in a browser this would be a must have.